.rubocop.yml

```diff
@@ -1,43 +1,113 @@
 
 Metrics/LineLength:
   Enabled: false
 
-Rails/TimeZone:
+Metrics/AbcSize:
   Enabled: false
 
-SignalException:
+################################################################################
+# Style
+################################################################################
+
+# Executables are conventionally named bin/foo-bar
+Style/FileName:
+  Exclude:
+  - bin/**/*
+
+# We don't (currently) document our code
+Style/Documentation:
   Enabled: false
 
+# Always use double-quotes to keep things simple
 Style/StringLiterals:
-  Enabled: false
+  EnforcedStyle: double_quotes
 
-Style/Documentation:
+# Use a trailing comma to keep diffs clean when elements are inserted or removed
+Style/TrailingComma:
+  EnforcedStyleForMultiline: comma
+
+# We avoid GuardClause because it can result in "suprise return"
+Style/GuardClause:
   Enabled: false
 
-Style/TrailingCommaInLiteral:
+# We avoid IfUnlessModifier because it can result in "suprise if"
+Style/IfUnlessModifier:
   Enabled: false
 
-Style/ClassAndModuleChildren:
+# We don't care about the fail/raise distinction
+Style/SignalException:
+  EnforcedStyle: only_raise
+
+Style/DotPosition:
+  EnforcedStyle: trailing
+
+# Common globals we allow
+Style/GlobalVars:
+  AllowedVariables:
+  - "$statsd"
+  - "$mongo"
+  - "$rollout"
+
+# Allow $! in config/initializers
+Style/SpecialGlobalVars:
   Exclude:
-    - 'spec/**/*'
+  - config/initializers/**/*
 
-Style/IfUnlessModifier:
+# We have common cases where has_ and have_ make sense
+Style/PredicateName:
+  Enabled: true
+  NamePrefixBlacklist:
+  - is_
+
+# We use %w[ ], not %w( ) because the former looks like an array
+Style/PercentLiteralDelimiters:
+  PreferredDelimiters:
+    "%w": []
+    "%W": []
+
+# Allow "trivial" accessors when defined as a predicate? method
+Style/TrivialAccessors:
+  AllowPredicates: true
+
+Style/Next:
   Enabled: false
 
-Style/DotPosition:
+# We think it's OK to use the "extend self" module pattern
+Style/ModuleFunction:
   Enabled: false
 
-Style/GuardClause:
+################################################################################
+# Rails - disable things because we're primarily non-rails
+################################################################################
+
+Rails/Delegate:
   Enabled: false
 
-Style/StringLiteralsInInterpolation:
+Rails/Output:
   Enabled: false
 
-Style/PercentLiteralDelimiters:
-  PreferredDelimiters:
-    '%w': []
-    '%W': []
+Rails/TimeZone:
+  Enabled: false
+
+################################################################################
+# Specs - be more lenient on length checks and block styles
+################################################################################
+
+Metrics/ModuleLength:
+  Exclude:
+  - spec/**/*
+
+Metrics/MethodLength:
+  Exclude:
+  - spec/**/*
+
+Style/ClassAndModuleChildren:
+  Exclude:
+  - spec/**/*
+
+Style/BlockDelimiters:
+  Exclude:
+  - spec/**/*
```

Comment:
We probably don't need these since we have the top level .codeclimate.yml spec excludes right?